shared/utils/replaceLinks.ts

```diff
@@ -7,7 +7,7 @@ const unicode = 'a-zA-Z0-9\\u00A0-\\uD7FF\\uF900-\\uFDCF\\uFDF0-\\uFFEF'
 const protocol = /([A-Za-z]{3,9}:(?:\/\/)?)/
 const hostname = /(?:[-;:&=+$,\w]+@)?([A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*)/
 // language=RegExp
-const path = `(?:\\/(?:[+~%/.${unicode}-]*[+~%/${unicode}-]+)?)?`
+const path = `(?:\\/(?:[+~%/_.${unicode}-]*[+~%/_${unicode}-]+)?)?`
```

Comment:
So the second underscore is for file name right?

Reply:
I think the second part of the regex is just ensuring that the last character is not a dot (`.`) and is therefore matches the extension (`pdf`) in this case, so the second underscore should actually not be needed.
```suggestion
const path = `(?:\\/(?:[+~%/_.${unicode}-]*[+~%/{unicode}-]+)?)?`
```